### ISSUE TITLE
High CPU Usage by Renderer Process in VSCode from Version 0.3.16

### ISSUE THREAD

**Issue by LangLangBart** [continuedev/continue#614]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- Continue: 0.5.0 (macOS Intel version)
- IDE: VSCode 1.84.2
```

### Description

A renderer process (`Code Helper (Renderer)`) in VSCode is **continuously** targeting ~100% of my CPU.

### bisect
- issue can first be noticed in version  **0.3.16**

```bash
#  0.3.16 Pre-Release  2023-11-08T22:23:27.323Z ✹ 08/Nov/23
#  0.3.15 Pre-Release  2023-11-08T08:47:22.21Z ✹ 08/Nov/23

git log --since=2023-11-08T08:47:22.21Z --until=2023-11-08T22:23:27.323Z --oneline
#  40067737 💚 Update package.json version [skip ci]
#  1fcc3f5c ci: 🏷 Update PyPI version [skip ci]
#  4fe45a9d 🐛 fix chunking bug
#  32aa2d24 ✨ warning if exceeding context limit
```

- issue relates to `ComboBox.tsx` => `useEffect` added with commit 32aa2d24
  - the `selectedContextItems` dependency updates **very** frequently

https://github.com/continuedev/continue/blob/32aa2d24ff5652477e4946c9cd021d616386e2c3/gui/src/components/ComboBox.tsx#L809-L817

---

PS: indexing was disabled `disable_indexing=True`

### To reproduce

1. Open VSCode and start a `New Session` (<kbd>⌥ Option</kbd> + <kbd>⌘ Command</kbd> + <kbd>N</kbd>) in `Continue`; execute a simple query, for example, "Hi".
2. Monitor the Activity Monitor for the `Code Helper (Renderer)` process, which is hitting at 100%.
  - Alternatively, you can also do it from the command line.
```bash
while :;do ps -A -o %cpu,comm | grep '/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)';date +"%T"; sleep 2;done
#   0.0 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)
#   0.0 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)
# 110.5 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)
# 16:02:13
#   0.0 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)
#   0.0 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)
# 101.2 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)
# 16:02:15
```

<img src="https://github.com/continuedev/continue/assets/92653266/ab0369e2-35fb-4dff-91ef-1241c0ed13f4" width="600">

3. Notice that when you disable the `Continue` extension the `Code Helper (Renderer)` process behaves normally.
  - Another way to verify this is by building the extension yourself and commenting out the lines listed here:

https://github.com/continuedev/continue/blob/32aa2d24ff5652477e4946c9cd021d616386e2c3/gui/src/components/ComboBox.tsx#L809-L817

**Comment by sestinj** [comment]
Changing the dependency to selectedContextItems.length fixes it


Weirdly reproducing this only required

```
  const selectedContextItems = useSelector((state: RootStore) => {
      return state.sessionState.contextItemsAtIndex[0] || []
  });
const [n, setN] = useState(0)
useEffect(() => {
setN(1)
}, [selectedContextItems])
```

And this happens even when I've made sure that none of the reducers that explicitly change contextItemsAtIndex are called.

**Comment by LangLangBart** [comment]
The fix is included in version **0.5.2**. Thank you for the rapid deployment.

---

```
  Version 5/475       Date (ISO timestamp ✹ relative)
  ¯¯¯¯¯¯¯¯¯¯¯¯¯       ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
  0.4.1               2023-11-10T20:40:29.797Z ✹ 20min ago
  0.5.2 Pre-Release   2023-11-10T20:02:30.66Z ✹ 58min ago
  0.5.0               2023-11-09T08:44:45.56Z ✹ 09/Nov/23
  0.3.18              2023-11-09T07:06:19.887Z ✹ 09/Nov/23
  0.3.17 Pre-Release  2023-11-09T03:52:21.33Z ✹ 09/Nov/23
```
